### PR TITLE
Frontend app

### PR DESCRIPTION
**Pull request by dinobi**

### CODE REVIEW

**Comment by andela-tezebuike** [comment]
Please fix houndCI issues

File: .eslintrc [hunk 1]
```diff
@@ -3,19 +3,36 @@
...
+  "plugins" : [
+    "react"
+  ],
   "rules": {
     "one-var": 0,
     "one-var-declaration-per-line": 0,
     "new-cap": 0,
     "consistent-return": 0,
-    "max-len": [1, 80, 2],
+    "max-len": [1, 100, 2],
```
Comment (tezebuike) [inline .eslintrc]: The Andela standard is usually 80. Please confirm if your LF agreed to 100

File: .coveralls.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+repo_token: 1LqYqgo8Da9hnzuSe1AV8P0SkIuilXTSu
```
Comment (tezebuike) [inline .coveralls.yml]: It is risky to include your token for people to use. You can add it on Travis settings

File: .gitignore [hunk 3]
```diff
@@ -2,20 +2,36 @@
...
 yarn-debug.log*
 yarn-error.log*
+materialize.min.css
+materialize.css
+materialize.min.js
+materialize.js
+roboto*
+/client/public/dist
+/.nyc_output
+/coverage
```
Comment (tezebuike) [inline .gitignore:27]: Awesome, Now delete the ones on your repo, commit and push the changes

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  - NODE_ENV=test
+before_script:
+  - psql -c 'create database travis_ci_test;' -U postgres
+  - sequelize db:migrate
+after_success:
+  - npm run coveralls
+after_script:
+  - nyc npm test && nyc report --reporter=text-lcov | coveralls
+notifications:
+  webhooks: https://coveralls.io/webhook?repo_token=1LqYqgo8Da9hnzuSe1AV8P0SkIuilXTSu
```
Comment (tezebuike) [inline .travis.yml:24]: Please remove token

File: .babelrc [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    // this plugin will be included only in development mode, e.g. 
+    // if NODE_ENV (or BABEL_ENV) environment variable is not set 
+    // or is equal to "development" 
+    "development": {
+      "plugins": [
+        "transform-decorators-legacy"
+      ]
+    }
+  }
+}
```
Comment (tezebuike) [inline .babelrc:13]: Always add a new line at the end.

File: README.md [hunk 6]
```diff
@@ -1,24 +1,48 @@
...
+- Post messages with added advantage of using wikipedia search right from you dashboard
+- Get messages on different medium based on message priority level
+- Have fun.
+
+# Installation and Dependencies
+> - Run `npm install` to install dependencies/dev-dependencies
+> - Setup Postgres
+> - setup your db with the settings in `server/config/config.json`
+> - Run `$ sequelize db:migrate`
+> - Run `npm start exec` to start server
```
Comment (tezebuike) [inline README.md:48]: Check comment for the README posted earlier

File: To-do [hunk 7]
```diff
@@ -0,0 +1,281 @@
+TO-DO
```
Comment (tezebuike) [inline To-do]: Remove file

File: client/.gitignore [hunk 8]
```diff
@@ -1,21 +1,21 @@
...
-.DS_Store
-.env.local
-.env.development.local
-.env.test.local
-.env.production.local
-
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
+# See https://help.github.com/ignore-files/ for more about ignoring files.
```
Comment (tezebuike) [inline client/.gitignore]: I think one gitignore is enough for the app

File: client/public/index.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<!doctype html>
```
Comment (tezebuike) [inline client/public/index.html]: The public folder should be auto generated. Hence is not needed in the repo

File: client/src/actions/create-group.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import actionType from '../actionTypes';
+import apiHandler from '../components/helpers/api-handler';
+
+export const onCreateGroupRequest = (groupDasta) => ({
```
Comment (tezebuike) [inline client/src/actions/create-group.js]: Please correct the spelling `groupDasta`

File: client/src/actions/create-group.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  type: actionType.CREATE_GROUP_REQUEST,
+  createGroupIsLoading: true
+});
+
+export const onCreateGroupSuccess = message => ({
+  type: actionType.CREATE_GROUP_SUCCESS,
+  creataGroupIsLoading: false,
```
Comment (tezebuike) [inline client/src/actions/create-group.js]: Spelling check `creataGroupIsLoading`

File: client/src/actions/create-group.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  groupsIsLoading: false,
+  message
+});
+
+const onCreateGroup = groupData =>
+(dispatch) => {
+  dispatch(onCreateGroupRequest(groupData));
+  let headers;
+  apiHandler('/api/create-group', groupData, 'post', headers).then((groupRes) => {
+    dispatch(onCreateGroupSuccess(groupRes.data));
```
Comment (tezebuike) [inline client/src/actions/create-group.js]: Best practice is to ensure variables are understood without assumption - `groupRes`

File: client/src/actions/create-group.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+  }).catch((groupRes) => {
+    dispatch(onCreateGroupFailure(groupRes.data));
+  });
+};
+
+export default onCreateGroup;
```
Comment (tezebuike) [inline client/src/actions/create-group.js]: Add a new line...always

**Pull request closed without merging**